### DISCUSSION TITLE
Best way to add a character counter to Controller wrapper component

### DISCUSSION THREAD

**Discussion by simplifyninja** [react-hook-form/react-hook-form#6248] [Q&A]
Hi there, I'm wondering what would be the best way to add a character counter to a Controller rendering as a StyledTextField?

Something that increases as a user types in the TextField like "15/100 characters".

I am able to add it to a standard TextField component using the code blocks below:

**JS**
```
const [values, setValues] = React.useState({ name: 'Sample description' });

const CHARACTER_LIMIT = 100;

const handleChange = (name: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
 setValues({ ...values, [name]: event.target.value });
};
```

**HTML**
```
<TextField
  label="Description"
  inputProps={{ maxLength: CHARACTER_LIMIT }}
  value={values.name}
  onChange={handleChange('name')}
  variant="outlined"
  margin="normal"
  fullWidth
/>
<StyledTextLabel>
  {`${values.name.length}/${CHARACTER_LIMIT}`} characters
</StyledTextLabel>
```

**Comment by bluebill1049** [comment]
use that component that you have and wrapped with `Controller`

**Reply by simplifyninja** [reply]
Thanks Bill. Do you have an example of the best way to wrap it?

I tried to use the render the TextField component but it doesn't seem to recognise field and onChange event.

Should I utilise useWatch() to subscribe to the input changes, similar to this CSB (but adapt it to the Controller)?

[https://codesandbox.io/s/react-hook-form-v7-ts-usewatch-forked-dbeyn?file=/src/index.tsx](https://codesandbox.io/s/react-hook-form-v7-ts-usewatch-forked-dbeyn?file=/src/index.tsx)

**Reply by simplifyninja** [reply]
I was able to resolve the issue utilising useWatch() on the TextField and returning the length of it.
